components/clp-package-utils/clp_package_utils/controller.py

```diff
@@ -0,0 +1,783 @@
+        env_vars |= {
+            "CLP_RESULTS_CACHE_STREAM_COLLECTION_NAME": (
+                self._clp_config.results_cache.stream_collection_name
+            ),
+        }
+        # Connection
+        env_vars |= {
+            "CLP_RESULTS_CACHE_DB_NAME": self._clp_config.results_cache.db_name,
+            "CLP_RESULTS_CACHE_HOST": _get_ip_from_hostname(self._clp_config.results_cache.host),
+            "CLP_RESULTS_CACHE_PORT": str(self._clp_config.results_cache.port),
+        }
+        # Path
+        env_vars |= {
+            "CLP_RESULTS_CACHE_CONF_FILE_HOST": str(conf_file),
+            "CLP_RESULTS_CACHE_DATA_DIR_HOST": str(data_dir),
+            "CLP_RESULTS_CACHE_LOGS_DIR_HOST": str(logs_dir),
+        }
+
+        return env_vars
+
+    def _set_up_env_for_compression_scheduler(self) -> EnvVarsDict:
+        """
+        Sets up environment variables and files for the compression scheduler component.
+
+        :return: Dictionary of environment variables necessary to launch the component.
+        """
+        component_name = COMPRESSION_SCHEDULER_COMPONENT_NAME
+        logger.info(f"Setting up environment for {component_name}...")
+
+        env_vars = EnvVarsDict()
+        # Logging
+        env_vars |= {
+            "CLP_COMPRESSION_SCHEDULER_LOGGING_LEVEL": (
+                self._clp_config.compression_scheduler.logging_level
+            ),
+        }
+
+        return env_vars
+
+    def _set_up_env_for_query_scheduler(self) -> EnvVarsDict:
+        """
+        Sets up environment variables and files for the query scheduler component.
+
+        :return: Dictionary of environment variables necessary to launch the component.
+        """
+        component_name = QUERY_SCHEDULER_COMPONENT_NAME
+        logger.info(f"Setting up environment for {component_name}...")
+
+        env_vars = EnvVarsDict()
+        # Logging
+        env_vars |= {
+            "CLP_QUERY_SCHEDULER_LOGGING_LEVEL": self._clp_config.query_scheduler.logging_level,
+        }
+
+        return env_vars
+
+    def _set_up_env_for_compression_worker(self, num_workers: int) -> EnvVarsDict:
+        """
+        Sets up environment variables for the compression worker component.
+
+        :param num_workers: Number of worker processes to run.
+        :return: Dictionary of environment variables necessary to launch the component.
+        """
+        component_name = COMPRESSION_WORKER_COMPONENT_NAME
+        logger.info(f"Setting up environment for {component_name}...")
+
+        logs_dir = self._clp_config.logs_directory / component_name
+        logs_dir.mkdir(parents=True, exist_ok=True)
+
+        env_vars = EnvVarsDict()
+        # Logging
+        env_vars |= {
+            "CLP_COMPRESSION_WORKER_LOGGING_LEVEL": (
+                self._clp_config.compression_worker.logging_level
+            ),
+        }
+        # Resource
+        env_vars |= {
+            "CLP_COMPRESSION_WORKER_CONCURRENCY": str(num_workers),
+        }
+
```

Comment:
I think we can just join these into one dictionary without comments since the grouping doesn't really improve readability in this case.